test-framework/oauth/src/main/java/org/keycloak/testframework/oauth/OAuthClientSupplier.java

```diff
@@ -52,12 +51,7 @@ public OAuthClient getValue(InstanceContext<OAuthClient, InjectOAuthClient> inst
 
     @Override
     public boolean compatible(InstanceContext<OAuthClient, InjectOAuthClient> a, RequestedInstance<OAuthClient, InjectOAuthClient> b) {
-        return true;
-    }
-
-    @Override
-    public LifeCycle getDefaultLifecycle() {
```

Comment:
This should still be global, no need to re-create the oauth client between test classes as long as it's compatible.